From: Alessandro Astone <alessandro.astone@canonical.com>
Date: Wed, 12 Feb 2025 14:09:26 +0100
Subject: fontdialogbutton: Gracefully handle unknown font

It should not be possible find no matching font-family after choosing from
the picker, but it is possible to set any font description with the
Gtk.FontDialogButton:font_desc property, e.g. when parsing from GtkBuilder.

The button already handles this case by showing "None" as its label.

(cherry picked from commit 7d05cd7771b843c1793177bae2efa36f983b85d5)

Origin: upstream, after 4.17.4
---
 gtk/gtkfontdialogbutton.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/gtk/gtkfontdialogbutton.c b/gtk/gtkfontdialogbutton.c
index 0eee4b3..1346544 100644
--- a/gtk/gtkfontdialogbutton.c
+++ b/gtk/gtkfontdialogbutton.c
@@ -621,6 +621,9 @@ update_font_data (GtkFontDialogButton *self)
         }
     }
 
+  if (self->font_family == NULL)
+    return;
+
   for (unsigned i = 0; i < g_list_model_get_n_items (G_LIST_MODEL (self->font_family)); i++)
     {
       PangoFontFace *face = g_list_model_get_item (G_LIST_MODEL (self->font_family), i);
